apps/server/src/modules/sync/events/workspace.ts

```diff
@@ -44,14 +44,18 @@ export class WorkspaceService {
   }
 
   async getDocFromGuid(guid: string): Promise<Doc | null> {
+    const updates = await this.storage.loadBuffer(guid);
+    if (!updates) return null;
+
     const doc = new Y.Doc({ guid });
     try {
-      // TODO load method return null if doc doesn't exist, error throwing is not needed.
-      const update = await this.storage.load(guid);
-      update && Y.applyUpdate(doc, update);
-      return doc;
+      updates.forEach(update => {
+        Y.applyUpdate(doc, update);
+      });
     } catch (e) {
-      return null;
+      console.error(e);
```

Comment:
I think we should return null here if update broke the yjs doc?

Reply:
emmmm, make sense, how do you think of this way?

[`d525500` (#3118)](https://github.com/toeverything/AFFiNE/pull/3118/commits/d525500187ba41d8dd401e0389bcccb66c70de32)

Reply:
I'm not sure how it is possible that an update cause an error.